tools/nanoplot/nanoplot.xml

```diff
@@ -0,0 +1,498 @@
+        <![CDATA[
+        #if $reads.type == 'pickle'
+            ln -s '$reads.files' './read.$reads.files.ext' &&
+        #else
+
+            reads='' &&
+            #for $i, $f in enumerate($reads.files)
+                #if $f
+                    #set $extension = $f.ext
+                    #if $extension == 'fastqsanger'
+                        #set $extension = 'fastq'
+                    #end if
+                    ln -s '$f' './read_${i}.$extension' &&
+                    reads="\$reads read_${i}.$extension" &&
+                #end if
+            #end for
+            reads=\$(echo \$reads | xargs) &&
+        #end if
+
+        NanoPlot
+            #if $reads.type == 'pickle'
+                --pickle './read.$reads.files.ext'
+            #else
+                --$reads.type \$reads
+            #end if
+
+            #if $filter.maxlength
+                --maxlength $filter.maxlength
+            #end if
+            #if $filter.minlength
+                --minlength $filter.minlength
+            #end if
+            #if $filter.downsample
+                --downsample $filter.downsample
+            #end if
+            #if $filter.minqual
+                --minqual $filter.minqual
+            #end if
+            #if $filter.readtype
+                --readtype $filter.readtype
+            #end if
+            #if $customization.color
+                --color $customization.color
+            #end if
+            #if $customization.format
+                --format $customization.format
+            #end if
+            #if $customization.plots
+                --plots ${str($customization.plots).replace(',', ' ')}
+            #end if
+            $filter.drop_outliers
+            $filter.loglength
+            $filter.percentqual
+            $filter.alength
+            $filter.barcoded
+            $customization.noN50
+            $customization.N50
+            #if $customization.title
+                --title '$customization.title'
+            #end if
+            $store
+            $raw
+            -o outputs &&
+            mv outputs/*.log outputs/NanoPlot_log.txt &&
+            if [[ -e outputs/NanoPlot-data.pickle ]]; then mv outputs/NanoPlot-data.pickle outputs/NanoPlot-data-pickle.binary; fi
+    ]]>
+    </command>
+    <inputs>
+        <conditional name="reads">
+            <param name="type" type="select" label="Type of the file(s) to work on">
+                <option value="fastq" selected="true">fastq</option>
+                <option value="fasta">fasta</option>
+                <option value="fastq_rich">fastq_rich</option>
+                <option value="fastq_minimal">fastq_minimal</option>
+                <option value="summary">summary</option>
+                <option value="bam">bam</option>
+                <option value="cram">cram</option>
+                <option value="pickle">pickle</option>
+            </param>
+            <when value="fastq">
+                <param
```

Comment:
Uha, now I get your gitter question.

I think a simple `<param
                    type="data"
                    name="files"
                    format="fasta,fasta.gz,fasta.bz2,cram,bam"
                    multiple="true" />`

should be enough. No need to support all file formats, just be most standard once and keep your life simple :)